### PR TITLE
ENG-450: TopBar revamp and restructure

### PR DESCRIPTION
**Pull request by ameer2468**
**Edit: Initially converted to draft**

- Based on my convos with Utku, we'll be shifting to this layout: < > ----- Search ------ Icons, 
- To address responsiveness, the tools/options, they will be collapsed as a burger menu (next PR)

**This is a restructure of the Topbar component, it is a now a component that is global across the app. With this new change here is what we got**:

- Modify the tools/options based on routes, some routes may require different options
- Tools/options can be grouped, upon grouping, within the UI there's a separator.
- TopBar.tsx is now a much cleaner file, previously it was filled with repetitive code.
- I installed framer-motion in preparation for minimal animations within the topbar for some search bar related animations. It's a great library for UI movement.

**Minor adjustments not relevant to this PR**:

- ScreenHeading component adjusted slightly and is now being used within all pages, as some pages were not using them. Overview didn't have a title, so I added it in there too.

**Whats next**:

- For responsiveness, the tools/options need to collapse.

**A video representing these changes:**

https://user-images.githubusercontent.com/33054370/231208329-7ef63378-e772-417a-aa14-86452d12dfab.mp4

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -52,5 +52,8 @@
 		"npm": "pnpm",
 		"yarn": "pnpm",
 		"node": ">=18.0.0"
+	},
+	"dependencies": {
+		"framer-motion": "^10.11.4"
```
Comment (utkubakir) [inline package.json:57]: This should be in `interface/package.json`

Comment (ameer2468) [reply package.json:57]: Will move it - i had doubts where to install it.

**Comment by ameer2468** [comment]
@jamiepine 

Spacing increased

Screenshot 2023-04-12 at 3 16 58 AM

**Comment by jamiepine** [comment]
Screenshot 2023-04-11 at 7 30 13 PM

These buttons should be grouped as they were before, by the looks of the code you might need to modify the TopBar's new API to support button grouping. Same goes for back/forward buttons, and probably other button groups in the future. Also, by spacing I meant between the divider, not each button. Should be an easy fix though.

Also, noticed the window is no longer draggable by the top bar, `data-tauri-drag-region` on outer div or the `DragRegion` component should be used here, somehow.

Love the look of it, and it makes a lot of sense to have this component on every primary screen.

**Comment by ameer2468** [comment]
> Screenshot 2023-04-11 at 7 30 13 PM
> 
> These buttons should be grouped as they were before, by the looks of the code you might need to modify the TopBar's new API to support button grouping. Same goes for back/forward buttons, and probably other button groups in the future. Also, by spacing I meant between the divider, not each button. Should be an easy fix though.
> 
> Also, noticed the window is no longer draggable by the top bar, `data-tauri-drag-region` on outer div or the `DragRegion` component should be used here, somehow.
> 
> Love the look of it, and it makes a lot of sense to have this component on every primary screen.

**For the sake of recording**:

Here are the updates that address this in a video:

https://user-images.githubusercontent.com/33054370/231413487-ec74a7f0-a96b-438a-94db-3c4139e5b643.mp4

**Pull request closed without merging**